ocaml/parsing/parser.mly

```diff
@@ -1887,18 +1887,24 @@ include_maybe_functor:
 
 (* A module type declaration. *)
 module_type_declaration:
-  MODULE TYPE
-  ext = ext
-  attrs1 = attributes
-  id = mkrhs(ident)
-  typ = preceded(EQUAL, module_type)?
-  attrs2 = post_item_attributes
-  {
-    let attrs = attrs1 @ attrs2 in
-    let loc = make_loc $sloc in
-    let docs = symbol_docs $sloc in
-    Mtd.mk id ?typ ~attrs ~loc ~docs, ext
-  }
+  | MODULE TYPE
+    ext = ext
+    attrs1 = attributes
+    id = mkrhs(ident)
+    typ = module_type_declaration_type
+    attrs2 = post_item_attributes
+    {
+      let attrs = attrs1 @ attrs2 in
+      let loc = make_loc $sloc in
+      let docs = symbol_docs $sloc in
+      Mtd.mk id typ ~attrs ~loc ~docs, ext
+    }
+;
```

Comment:
After refactoring, this rule has only one line of diff. We want to minimizes diffs, so that in the long run the origin of code is easier to track.

Reply:
Great point - thank you!